gazelle/tests/integrationtests/com/google/idea/blaze/gazelle/sync/RunGazelleOnSyncTest.java

```diff
@@ -0,0 +1,184 @@
+                "gazelle(name = \"gazelle\")"
+        );
+        workspace.createFile(new WorkspacePath("WORKSPACE"),
+                "load(\"@bazel_tools//tools/build_defs/repo:http.bzl\", \"http_archive\")",
+                "",
+                "http_archive(",
+                "    name = \"io_bazel_rules_go\",",
+                "    sha256 = \"2b1641428dff9018f9e85c0384f03ec6c10660d935b750e3fa1492a281a53b0f\",",
+                "    urls = [",
+                "        \"https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.29.0/rules_go-v0.29.0.zip\",",
+                "        \"https://github.com/bazelbuild/rules_go/releases/download/v0.29.0/rules_go-v0.29.0.zip\",",
+                "    ],",
+                ")",
+                "",
+                "http_archive(",
+                "    name = \"bazel_gazelle\",",
+                "    sha256 = \"de69a09dc70417580aabf20a28619bb3ef60d038470c7cf8442fafcf627c21cb\",",
+                "    urls = [",
+                "        \"https://mirror.bazel.build/github.com/bazelbuild/bazel-gazelle/releases/download/v0.24.0/bazel-gazelle-v0.24.0.tar.gz\",",
+                "        \"https://github.com/bazelbuild/bazel-gazelle/releases/download/v0.24.0/bazel-gazelle-v0.24.0.tar.gz\",",
+                "    ],",
+                ")",
+                "",
+                "load(\"@io_bazel_rules_go//go:deps.bzl\", \"go_register_toolchains\", \"go_rules_dependencies\")",
+                "load(\"@bazel_gazelle//:deps.bzl\", \"gazelle_dependencies\", \"go_repository\")",
+                "go_rules_dependencies()",
+                "",
+                "go_register_toolchains(version = \"1.17.2\")",
+                "",
+                "gazelle_dependencies()"
+        );
+        workspace.createDirectory(new WorkspacePath("src"));
+        workspace.createFile(new WorkspacePath("src/main.go"),
+                "package main",
+                "import \"fmt\"",
+                "func main() {",
+                "  fmt.Println(\"Hello World\")",
+                "}"
+        );
+    }
+
+    private void runBazelSync() {
+        BlazeSyncParams syncParams =
+                BlazeSyncParams.builder()
+                        .setTitle("Full Sync")
+                        .setSyncMode(SyncMode.FULL)
+                        .setSyncOrigin("test")
+                        .setAddProjectViewTargets(true)
+                        .build();
+        runBlazeSync(syncParams);
+    }
+
+    @Test
+    public void testGazelleDoesntRunWhenNotConfigured() throws Exception {
+        setUpRepoWithGazelle();
+        setProjectView(
+                "directories:",
+                "  ."
+        );
+
+        runBazelSync();
+
+        errorCollector.assertNoIssues();
+
+        File resultBuildFile = workspaceRoot.fileForPath(new WorkspacePath("src/BUILD.bazel"));
+        assertThat(workspaceRoot.isInWorkspace(resultBuildFile)).isFalse();
```

Comment:
I'm not sure but I think `workspaceRoot.isInWorkspace()` will compare the path of the file to the path of the workspace directory instead of checking the existence of the build file which we want here. So if you can confirm that, we can replace this line with `assertThat(resultBuildFile.exists()).isFalse();`  
Same for the other tests.

Reply:
Good catch, you were right. I've changed it for that test, and added the actual existence check for the other tests.